game/functions/core/client/statusEffect/fn_statusEffect_set.sqf

```diff
@@ -47,10 +59,21 @@ if (isNil "_effectsMap") then {
     }]
 };
 
-format ["Adding status effect reason: %1 | %2", _effect, _reason] call vgm_g_fnc_logDebug;
+format ["Adding status effect reason: %1 | %2 | %3", _effect, _reason, _duration] call vgm_g_fnc_logDebug;
 
 private _reasonList = _effectsMap getOrDefault [_effect, [], true];
 
+private _effectsReasonEndTimes = _effectsEndTimes getOrDefault [_effect, createHashMap, true];
+if (_duration > 0) then {
+    private _endTime = serverTime + _duration;
```

Comment:
I would use `time` which is tracked since mission started and affected by accTime in SP (does not matter for us though).
You usually want to avoid server time as it is more prone to floating point errors.

```suggestion
    private _endTime = time + _duration;
```